app/src/main/java/org/mifos/mobile/ui/update_password/UpdatePasswordScreen.kt

```diff
@@ -0,0 +1,253 @@
+import androidx.compose.ui.text.input.VisualTransformation
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import kotlinx.coroutines.launch
+import org.mifos.mobile.R
+import org.mifos.mobile.core.ui.component.MifosOutlinedTextField
+import org.mifos.mobile.core.ui.component.MifosProgressIndicator
+import org.mifos.mobile.core.ui.component.MifosTopBar
+import org.mifos.mobile.core.ui.theme.DarkPrimary
+import org.mifos.mobile.core.ui.theme.LightPrimary
+import org.mifos.mobile.core.ui.theme.MifosMobileTheme
+import org.mifos.mobile.ui.savings_account_withdraw.UiStatesParameterProvider
+import org.mifos.mobile.utils.RegistrationUiState
+
+@SuppressLint("CoroutineCreationDuringComposition")
+@OptIn(ExperimentalComposeUiApi::class, ExperimentalMaterial3Api::class)
+@Composable
+fun UpdatePasswordScreen(
+    viewModel: UpdatePasswordViewModel = hiltViewModel(),
+    getBackToPreviousScreen: () -> Unit
+) {
+    var newPassword by rememberSaveable {
+        mutableStateOf("")
+    }
+    var confirmPassword by rememberSaveable {
+        mutableStateOf("")
+    }
+    var viewNewPassword by rememberSaveable {
+        mutableStateOf(false)
+    }
+    var viewConfirmPassword by rememberSaveable {
+        mutableStateOf(false)
+    }
+
+    var newPasswordError by rememberSaveable { mutableStateOf(false) }
+    var confirmPasswordError by rememberSaveable { mutableStateOf(false) }
+
+    var newPasswordErrorContent by rememberSaveable {
+        mutableStateOf("")
+    }
+    var confirmPasswordErrorContent by rememberSaveable {
+        mutableStateOf("")
+    }
+
+    val snackbarHostState = remember { SnackbarHostState() }
+    val scope = rememberCoroutineScope()
+
+    val uiState by viewModel.updatePasswordUiState.collectAsStateWithLifecycle()
+    val keyboardController = LocalSoftwareKeyboardController.current
+    val context = LocalContext.current
+
+    Scaffold(
+        snackbarHost = {
+            SnackbarHost(hostState = snackbarHostState)
+        },
+        topBar = {
+            MifosTopBar(title = { Text(text = stringResource(id = R.string.change_password)) }, navigateBack = getBackToPreviousScreen)
+        }
+    ) {
+        Box {
+            when (uiState) {
+                is RegistrationUiState.Loading -> {
+                    AlertDialog(onDismissRequest = {}) { MifosProgressIndicator(modifier = Modifier.fillMaxSize()) }
+                }
+
+                is RegistrationUiState.Error -> {
+                    scope.launch {
+                        snackbarHostState.showSnackbar(
+                            context.getString(R.string.could_not_update_password_error), context.getString(R.string.dialog_action_ok), duration = SnackbarDuration.Short
```

Comment:
i think the default duration of showSnackbar is `SnackbarDuration.Short` so this is redundant. Please check